drivers/soundwire/stream.c

```diff
@@ -1752,17 +1805,31 @@ int sdw_deprepare_stream(struct sdw_stream_runtime *stream)
 {
 	int ret;
 
+	pr_err("%s: %s: start\n", __func__, stream->name);
+
 	if (!stream) {
 		pr_err("SoundWire: Handle not found for stream\n");
 		return -EINVAL;
 	}
 
 	sdw_acquire_bus_lock(stream);
+
+	if (stream->state != SDW_STREAM_DISABLED) {
```

Comment:
@plbossart prepared <--> deprepared is valid, right? I can see hw_parames -> prepare -> hw_free without ant trigger when pulseaudio on. I.e. No enable/disable between prepare/deprepare. I think it makes sense since the application may only want to configure the stream but not to play it. So I think we can add && stream->state != SDW_STREAM_PREPARED here.

Reply:
You are right, but that requires an update to the state machine as well. Will fix.
Nice catch, thanks @bardliao